source/adapters/level_zero/v2/command_buffer.cpp

```diff
@@ -0,0 +1,309 @@
+      IsUpdatable(Desc ? Desc->isUpdatable : false) {
+  UR_CALL_THROWS(ur::level_zero::urContextRetain(Context));
+  UR_CALL_THROWS(ur::level_zero::urDeviceRetain(Device));
+}
+
+void ur_exp_command_buffer_handle_t_::cleanupCommandBufferResources() {
+  // Release the memory allocated to the Context stored in the command_buffer
+  UR_CALL_THROWS(ur::level_zero::urContextRelease(Context));
+
+  // Release the device
+  UR_CALL_THROWS(ur::level_zero::urDeviceRelease(Device));
+
+  // Release the memory allocated to the CommandList stored in the
+  // command_buffer
+  if (ZeCommandList) {
+    ZE_CALL_NOCHECK(zeCommandListDestroy, (ZeCommandList));
+  }
+
+  for (auto &AssociatedKernel : KernelsList) {
+    UR_CALL_THROWS(ur::level_zero::urKernelRelease(AssociatedKernel));
+  }
+}
+
+namespace ur::level_zero {
+
+/**
+ * Creates a L0 command list
+ * @param[in] Context The Context associated with the command-list
+ * @param[in] Device  The Device associated with the command-list
+ * @param[in] IsUpdatable Whether the command-list should be mutable.
+ * @param[out] CommandList The L0 command-list created by this function.
+ * @return UR_RESULT_SUCCESS or an error code on failure
+ */
+ur_result_t createMainCommandList(ur_context_handle_t Context,
+                                  ur_device_handle_t Device,
+                                  bool IsUpdatable,
+                                  ze_command_list_handle_t &CommandList) {
+
+
+  using queue_group_type = ur_device_handle_t_::queue_group_info_t::type;
+  // that should be call to queue getZeOrdinal, 
+  // but queue is not available while constructing buffer
+  uint32_t QueueGroupOrdinal = Device->QueueGroup[queue_group_type::Compute].ZeOrdinal;
+
+  ZeStruct<ze_command_list_desc_t> ZeCommandListDesc;
+  ZeCommandListDesc.commandQueueGroupOrdinal = QueueGroupOrdinal;
+
+  ZeCommandListDesc.flags = ZE_COMMAND_LIST_FLAG_IN_ORDER;
+
+  ZeStruct<ze_mutable_command_list_exp_desc_t> ZeMutableCommandListDesc;
+  if (IsUpdatable) {
+    ZeMutableCommandListDesc.flags = 0;
+    ZeCommandListDesc.pNext = &ZeMutableCommandListDesc;
+  }
+
+  ZE2UR_CALL(zeCommandListCreate, (Context->getZeHandle(), Device->ZeDevice,
+                                   &ZeCommandListDesc, &CommandList));
+
+  return UR_RESULT_SUCCESS;
+}
+
+ur_result_t
+urCommandBufferCreateExp(ur_context_handle_t Context, ur_device_handle_t Device,
+                         const ur_exp_command_buffer_desc_t *CommandBufferDesc,
+                         ur_exp_command_buffer_handle_t *CommandBuffer) {
+  bool IsUpdatable = CommandBufferDesc && CommandBufferDesc->isUpdatable;
+  checkImmediateAppendSupport(Context, Device);
+
+  if (IsUpdatable) {
+    UR_ASSERT(Context->getPlatform()->ZeMutableCmdListExt.Supported,
+              UR_RESULT_ERROR_UNSUPPORTED_FEATURE);
+  }
+  
+  ze_command_list_handle_t ZeCommandList = nullptr;
+  UR_CALL(createMainCommandList(Context, Device, IsUpdatable, ZeCommandList));
+  try {
```

Comment:
please move the try/catch block to cover the entire function and add try/catch to other functions as well, i.e. `urCommandBufferCreateExp(...) try { ... } catch (...) {  return exceptionToResult(std::current_exception()); }` In v2, we have helpers functions that might throw so it's best to wrap every function with try/catch.